packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -436,20 +439,19 @@ export async function sourceNodes(
 
   if (assets.length) {
     reporter.info(`Creating ${assets.length} Contentful asset nodes`)
-  }
-
-  for (let i = 0; i < assets.length; i++) {
-    // We wait for each asset to be process until handling the next one.
-    await Promise.all(
-      createAssetNodes({
-        assetItem: assets[i],
-        createNode,
-        createNodeId,
-        defaultLocale,
-        locales,
-        space,
-      })
-    )
+    for (let i = 0; i < assets.length; i++) {
+      // We wait for each asset to be process until handling the next one.
+      await Promise.all(
+        createAssetNodes({
+          assetItem: assets[i],
+          createNode,
+          createNodeId,
+          defaultLocale,
+          locales,
+          space,
+        })
+      )
+    }
```

Comment:
This change could/should also happen before this major. Just a cleanup/restructure.